nemo/collections/asr/data/audio_to_audio.py

```diff
@@ -314,7 +324,18 @@ def process_audio(self, audio: Dict[str, torch.Tensor]) -> Dict[str, torch.Tenso
         Returns:
             An ordered dictionary of signals and their tensors.
         """
-        # Currently, not doing any processing of the loaded signals.
+        if self.normalize:
+            # Normalize audio w.r.t. to the first signal
+            norm_scale = audio[self.normalization_signal].abs().max()
+
+            # Do not normalize embeddings
+            skip_signals = self.embedding_setup.signals if self.embedding_setup is not None else []
+
+            # Normalize audio signals
+            for signal in audio:
+                if signal not in skip_signals:
+                    audio[signal] = audio[signal] / (norm_scale + self.eps)
```

Comment:
As I understand this code, it assumes that all `signal`s in `audio` are on the same scale to begin with, and we are just rescaling them by the same factor.

Shouldn't normalization depend on some signal statistic? `/= audio[signal] / audio[signal].abs().max() * (norm_scale + self.eps)` or something along these lines?

Reply:
All makes sense as long as the user understands this is about paired data

Thanks for adding the comment